src/main/java/org/jenkinsci/plugins/tokenmacro/impl/BuildLogRegexMacro.java

```diff
@@ -234,15 +240,17 @@ String getContent(BufferedReader reader)
                 }
             }
             if (numLinesTruncated > 0) {
-                insidePre = stopPre(buffer, insidePre);
-                appendLinesTruncated(buffer, numLinesTruncated, asHtml);
+                insidePre = stopPre(matchResults, insidePre);
+                appendLinesTruncated(matchResults, numLinesTruncated, asHtml);
             }
         }
-        insidePre = stopPre(buffer, insidePre);
-        if (buffer.length() == 0) {
+        insidePre = stopPre(matchResults, insidePre);
+        if (matchResults.size() == 0) {
             return defaultValue;
         }
-        return buffer.toString();
+        if (maxTailMatches > 0 && matchResults.size() > maxTailMatches) {
+            matchResults = matchResults.subList(matchResults.size() - maxTailMatches, matchResults.size());

```

Comment:
What happens if it is in the middle of a `<pre></pre>` and cuts off? See the insidePre checks above. I think we would want to make sure that a `</pre>` gets added at the end if one is needed.

Reply:
Thanks for pointing that out, let me look into the right handling for this case.

Reply:
I addressed your concern, please let me know if you have any other feedback.